src/Adapters/EventSource.Netstandard13/EventSourceTelemetryModule.cs

```diff
@@ -176,6 +192,21 @@ protected override void OnEventSourceCreated(EventSource eventSource)
         }
 
         /// <summary>
+        /// Invokes when change happens in DisabledSources list.
+        /// </summary>
+        private void DisabledEventSources_CollectionChanged(object sender, System.Collections.Specialized.NotifyCollectionChangedEventArgs e)
```

Comment:
My understanding is reacting to configuration changes in real time is not really expected from telemetry modules. What is expected is reacting to configuration changes when ITelemetryModule.Initialize() is called.

@SergeyKanzhelev do you have a preference?

@xiaomi7732 This scenario warrants a unit test (one that calls Initialize() twice and verifies that the behavior has changed according to changed configuration)

Reply:
@karolz-ms,
The intention was to save the calculation when checking if there have been disable-listening requests.
Now I agree that it a better idea to have a snap of the decision than the current implementation. I'll update the code in the next iteration.